zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/Collection.java

```diff
@@ -478,66 +474,57 @@ public static Collection update(
         Set<String> teamsUpdated = new HashSet<>();
 
         if (desc != null && desc.getTeams() != null) {
-            UsersService users = zebedee.getUsersService();
             PermissionsService permissions = zebedee.getPermissionsService();
-            TeamsService teams = zebedee.getTeamsService();
-            CollectionKeyring collectionKeyring = zebedee.getCollectionKeyring();
-
 
             // Remove any teams that should no longer have access.
-            removeTeamsFromCollection(teams, users, permissions, collectionKeyring, desc, session);
+            removeTeamsFromCollection(permissions, desc, session);
 
             // Add any new teams that have been granted access.
-            teamsUpdated.addAll(addTeamsToCollection(teams, users, permissions, collectionKeyring, desc, session));
+            // TODO: Remove dependency on the deprecated {@link TeamsService}. See addTeamsToCollection below for more details.
+            teamsUpdated.addAll(addTeamsToCollection(zebedee.getTeamsService(), permissions, desc, session));
         }
 
         return teamsUpdated;
     }
 
     /**
-     * Remove any Teams from the collection if it is no longer assigned. For each team that is removed the collection
-     * key is revoked from each member of the team.
+     * Remove any Teams from the collection if it is no longer assigned.
      */
-    private static void removeTeamsFromCollection(TeamsService teams,
-                                                  UsersService users, PermissionsService permissions, CollectionKeyring collectionKeyring,
-                                                  CollectionDescription desc, Session session)
+    private static void removeTeamsFromCollection(PermissionsService permissions, CollectionDescription desc,
+                                                  Session session)
             throws IOException, ZebedeeException {
         Set<Integer> teamIDs = permissions.listViewerTeams(desc, session);
 
-        List<Team> teamsToRemove = teams.resolveTeams(teamIDs)
-                .stream()
-                .filter(t -> !desc.getTeams().contains(t.getId()))
+        List<Integer> teamsToRemove = teamIDs.stream().filter(t -> !desc.getTeams().contains(t))
```

Comment:
This is currently only called from update collection, which is only called from PUT requests to the collection endpoint. The list of teams in the `CollectionDescription` object is from this request body and is therefore the list of teams that the user has specified via Florence. This means that we can trust the list since this is the list the user is expecting. Even if the update method gets called in a different context in future (which should be unlikely since all of this logic will be removed soon), the caller will always be providing collection description so we shouldn't be relying on the collection details read from the collection JSON.

Reply:
The potential problem is that the data in the user browser might be out of date with the server. For example if you make changes to the collection but before you save that update I also GET the page and make a change when you save your changes you will now overwrite any changes I've made without realising it.

It's an unavoidable flow with the current design of the CMS and not one I think we can fix but I wanted to highlight it as a potential problem/complication.

Reply:
I agree there is a high risk of race conditions between client read and write. Ideally we would implement eTag and If-Match headers for our API calls, but since this code will be transitional and short lived we will leave it for now.